### DISCUSSION TITLE
Error when creating 2nd level page naming with characters identical to preserved path

### DISCUSSION THREAD

**Discussion by kaxing** [requarks/wiki#3603] [Error / Bug Report]
**Describe the bug**

There are two related issues that I found when testing around Multilingual Namespacing, please let me know if I need to open another issue for the last one.

1. Unable to create the page with the same or preserved keywords after 2nd level paths:
`en/home/en` works, but `en/en/home` does not and the error says "Cannot create this page because it starts with a system reserved path."

2. Unsupported or disabled language(namespace) will still able to create the page but the UI will display in untranslated keys.

**To Reproduce**
1. Open the browser
2. Type in URL bar `domain.name/zz/zz/home` opens in 'page not exit' view
3. Click on 'newpage.create'

**Expected behavior**
Be able to create a new page using any naming after 2nd level of the path, as the preserved language code should be limited to 1st level path.
Also, the UI should fallback to the default language or present certain messages when there is no locale or not enabled.

**Screenshots**
for step 2 and 3:
![image](https://user-images.githubusercontent.com/602540/89381139-4803de00-d72b-11ea-83f1-0b35cc2c8242.png)
![image](https://user-images.githubusercontent.com/602540/89379629-b4311280-d728-11ea-94bf-a13578107c34.png)

**Host Info (please complete the following information):**
 - OS: Ubuntu 18.04 lxd/lxc 20.04
 - Wiki.js version: 2.4.107
 - Database engine: sqlite 3.31.1

**Additional context**
Let me know if I missed out anything, thank you.

**Comment by 3DSkaoi** [comment]
Try changing your path 

- OLD: domain.name/zz/zz/home
- NEW: domain.name/zzz/zz/home 

As I understand WIKI.JS is not checking if those 2 letters are locale and prevent you from creating the page.

**Comment by kaxing** [comment]
Hey @3DSkaoi thanks!

By using your example with Multilingual Namespacing enabled,
it will end up with the site default language, for example in English:
- domain.name/en/zzz/zz/home

And the problem here is I cannot create page like:
- domain.name/en/fr/home
or even:
- domain.name/en/en-us/home

**Comment by Smankusors** [comment]
eh... uh.... why would you create path like that? 🤔

**Comment by kaxing** [comment]
I was trying to transfer some articles from one service to wiki.js.
My original plan is to replace the hosting with the same domain name and keep the entire article URLs structure.

However, there is a folder name `hc` right after the root path in each article URL, that is how I step into this problem. And `hc` happen to be a reserved language code, these hc-URLs will be invalid with Multilingual Namespacing enabled.

**Comment by Smankusors** [comment]
uh oh... Is your Wiki.js behind reverse proxy? Perhaps you can setup redirect the old URL to the new URL instead?

**Comment by AMDphreak** [comment]
I got this same error by renaming the path of an existing article from something to `/to-do`. It wouldn't let me link the page in the navigation bar, which was weird. So I tried editing the page and that produced the error page. There is no way to edit the page to undo the mistake, so I had to create a new page and copy the existing one to it, then delete the old page.
